### PR TITLE
updated the inventory

### PR DESCRIPTION
**Pull request by msavya**
# Pull Request

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Refactoring (no functional changes)
- [ ] Performance improvement
- [ ] Other (please describe):

## Related Issues (put task name here from notion)
Updated Inventory & pending modal
NEW: Inventory Modal

### Code Review Preparation
- [ ] I have performed a self-review and reviewed the changed files
- [ ] My code follows the project's style guidelines and passes linting
- [ ] I have added comments in hard-to-understand areas and updated documentation

### Testing (Ignore for the first PR)
- [ ] I have added relevant tests and they pass locally
- [ ] I have tested the changes manually, including edge cases

### Code Quality & Security
- [ ] I have formatted my code and removed debugging/unused code
- [ ] No TypeScript errors or linting issues
- [ ] No sensitive information committed (API keys, passwords, etc.)
- [ ] User inputs are validated and sanitized

### Dependencies & Deployment
- [ ] Dependencies are documented and package-lock.json is updated if needed
- [ ] Changes work with Firebase emulator and environment variables are documentedRetry

## Screenshots (If it is a front end feature screenshot is required)
Screenshot 2026-03-23 at 6 20 37 PM
Screenshot 2026-04-06 at 1 55 31 AM
Screenshot 2026-04-06 at 1 55 45 AM

## Release Notes

* **New Features**
  * Inventory page is now fully functional with search, status filtering, and sorting capabilities
  * Added visual dials displaying inventory quantities with color-coded stock status indicators
  * Implemented two-step modal workflow for editing and confirming inventory count changes

### CODE REVIEW

File: app/inventory/layout.tsx [hunk 1]
```diff
@@ -2,24 +2,52 @@
...
+import Link from "next/link";
+import { usePathname } from "next/navigation";
 import { ReactNode } from "react";
 
 export default function InventoryLayout({ children }: { children: ReactNode }) {
+    const pathname = usePathname();
+
     return (
         <ProtectedRoute allow={["Admin"]}>
-            <div className="h-full w-full flex flex-col font-family-roboto overflow-hidden">
```
Comment (joelchem) [inline app/inventory/layout.tsx]: maintain the old styling here, it ensures scrolling works properly within the page so that the entire page is not extended when there is too much content

File: app/inventory/layout.tsx [hunk 2]
```diff
@@ -2,24 +2,52 @@
...
-                <div className="flex flex-1 min-h-0">
+            <div className="h-full w-full flex flex-col font-family-roboto">
+                <div className="flex flex-1">
                     <SideNavbar />
-                    <div className="flex-1 min-h-0 bg-[#F7F7F7] pt-8 pb-4 px-6 flex flex-col">
+                    <div className="flex-1 bg-[#F7F7F7] py-8 px-16 flex flex-col">
```
Comment (joelchem) [inline app/inventory/layout.tsx]: keep the page padding consistent with the other pages. i know its different on the figma now, but for now its easier to keep it consistent with the pages that are already there, and if we want to change it in the future we can change it for all pages

File: app/inventory/layout.tsx [hunk 3]
```diff
@@ -2,24 +2,52 @@
...
                         <span className="text-2xl text-primary font-extrabold block">
                             Inventory
                         </span>
-                        <div className="bg-background rounded-xl my-2 flex-1 py-4 px-6 min-h-0 overflow-hidden flex flex-col">
+                        <div className="flex gap-8 text-sm">
```
Comment (joelchem) [inline app/inventory/layout.tsx]: don't need the links here. the changelog was moved so the inventory page is independent again

File: components/inventory/CategoryDialsGrid.tsx [hunk 4]
```diff
@@ -0,0 +1,22 @@
+"use client";
+
+import { InventoryCategory } from "@/types/inventory";
+import { ItemDial } from "./ItemDial";
+
+export function CategoryDialsGrid({ categories }: { categories: InventoryCategory[] }) {
```
Comment (joelchem) [inline components/inventory/CategoryDialsGrid.tsx]: the category dials grid should show the category dials in the same pattern as how the pickups/deliveries dashboard shows pickups/deliveries and how the (old) gallery view showed the gallery view cards. As the screen size gets larger it should add more items in each row to maximize space used. I think the easiest place to reference this is the GalleryView component or the pickups/deliveries stuff mentioned earlier

**Review by joelchem — CHANGES_REQUESTED** [review]
Great job overall with this PR! 

There are some small styling changes and random other additions that we want to see which is what most of the comments are about. As always, let us know if you have any questions!

**Review by joelchem — APPROVED** [review]
Looks great! Made some refactors and going ahead and merging it in!
